packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -3128,19 +3110,107 @@ export class StripeHelper extends StripeHelperBase {
 
     const productPaymentCycleOld = this.stripePlanToPaymentCycle(planOld);
 
+    const sameBillingCycle =
+      productPaymentCycleOld === baseDetails.productPaymentCycleNew;
+
+    // get next invoice details
+    const nextInvoice = upcomingInvoiceWithInvoiceItem
+      ? upcomingInvoiceWithInvoiceItem
+      : await this.previewInvoiceBySubscriptionId({
+          subscriptionId: subscription.id,
+        });
+
+    const {
+      total: nextInvoiceTotal,
+      currency: nextInvoiceCurrency,
+      lines: nextInvoiceLines,
+    } = nextInvoice || {};
+
+    // https://stripe.com/docs/api/invoices/object#invoice_object-lines
+    // as per the Stripe docs, one of the individual line items includes
+    // new plan pricing and tax information without additional fees
+    const upcomingInvoiceLineItem =
+      nextInvoiceLines && nextInvoiceLines.data
+        ? nextInvoiceLines.data.find((line) => !line.proration)
+        : undefined;
```

Comment:
This can be refactored to the following.
```
const upcomingInvoiceLineItem = nextInvoiceLines?.data?.find((line) => !line.proration)
```

Reply:
Good catch! Updated!